src/index.ts

```diff
@@ -32,13 +32,33 @@ const schema = new GraphQLSchema({
       },
       block: {
         type: Block,
-        args: { number: { type: GraphQLInt } },
-        resolve: (obj, { blockNumber }) => web3.eth.getBlock(blockNumber, true),
+        args: { number: { type: GraphQLInt }, hash: { type: GraphQLString } },
+        resolve: (obj, { block_number, hash }) => {
+          if ((!hash && !block_number) || (hash && block_number)) {
+            throw new Error('Please provide one argument.');
+          }
+          return block_number ? web3.eth.getBlock(block_number, true) : web3.eth.getBlock(hash, true);
+        },
       },
       blocks: {
         type: new GraphQLList(Block),
-        args: { from: { type: GraphQLInt }, to: { type: GraphQLInt } },
-        resolve: (obj, { from, to }) => Promise.all(_.range(from, to + 1).map(i => web3.eth.getBlock(i, true))),
+        args: {
+          from: { type: GraphQLInt },
+          to: { type: GraphQLInt },
+          numbers: { type: new GraphQLList(GraphQLInt) },
+          hashes: { type: new GraphQLList(GraphQLString) },
+        },
+        resolve: (obj, { from, to, hashes, numbers }) => {
+          if (hashes && !numbers && !from && !to) {
```

Comment:
Ideally we would return different messages depending on the validation error. We should run all validations at the top of the function. The way I would approach it is:

1. If `hashes` is not null but has length zero, locally set it to undefined to facilitate validations. Same for `numbers`.
2. If `from` is "", set it to undefined to facilitate validations. Same for `to`.
3. Check that all 4 args are not undefined. => message example: "Please provide either: (1) both 'from' and 'to', (2) a list of block hashes in 'hashes', or (3) a list of block numbers in 'numbers'."
4. If both `from` or `to` are not undefined, check that both `numbers` and `hashes` are.
5. If `numbers` is not undefined, check that all other 3 are.
6. If `hashes` is not undefined, check that all other 3 are.

Then we can run a 3-part if/else statement that covers each variant.